src/Microsoft.SqlTools.ServiceLayer/LanguageServices/AutoCompleteHelper.cs

```diff
@@ -728,5 +728,137 @@ internal static SignatureHelp ConvertMethodHelpTextListToSignatureHelp(List<Babe
 
             return help;
         }
+
+        /// <summary>
+        /// Give suggestions for sql star expansion. 
+        /// </summary>
+        /// <param name="scriptDocumentInfo">Document info containing the current cursor position</param>
+        /// <returns>Completion item array containing the expanded star suggestion</returns>
+        public static CompletionItem[] ExpandSqlStarExpression(ScriptDocumentInfo scriptDocumentInfo)
+        {
+            //Fetching the star expression node in sql script.
+            Microsoft.SqlServer.Management.SqlParser.SqlCodeDom.SqlSelectStarExpression selectStarExpression = AutoCompleteHelper.isSelectStarStatement(scriptDocumentInfo.ScriptParseInfo.ParseResult.Script, scriptDocumentInfo);
+            if (selectStarExpression == null)
+            {
+                return null;
+            }
+
+            // Getting SQL object identifier for star expressions like a.* 
+            Microsoft.SqlServer.Management.SqlParser.SqlCodeDom.SqlObjectIdentifier starObjectIdentifier = null;
+            if (selectStarExpression.Children.Count() != 0)
+            {
+                starObjectIdentifier = (Microsoft.SqlServer.Management.SqlParser.SqlCodeDom.SqlObjectIdentifier)selectStarExpression.Children.ElementAt(0);
+            }
+            /*
+                Getting bounded tables associated with the star query. This information will help us form column names with their parent tables when multiple tables are present in the query.
+                Currently, SqlParser does not publicly expose bounded table. Therefore, using reflection to get bounded tables. 
+            */
+            // 
+            Object bindingContext = typeof(Microsoft.SqlServer.Management.SqlParser.SqlCodeDom.SqlCodeObject).GetProperty("BindingContext", System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance).GetValue(selectStarExpression);

```

Comment:
Was this logic you wrote yourself or did you copy this from somewhere? Why are you using reflection here?

Reply:
But also - I would suggest looking at how SSMS does it since that's an already working example that has gone through a lot of testing over the years already.

Reply:
Adding @Matteo-T and @shueybubbles to this PR, to see if there is a better way to do this.